packages/hyperion-flowlet/src/FlowletManager.ts

```diff
@@ -146,9 +147,56 @@ export class FlowletManager<T extends Flowlet = Flowlet> {
       return listener;
     }
 
-    const funcInterceptor = interceptEventListener(listener);
-    if (funcInterceptor && !funcInterceptor.testAndSet(IS_FLOWLET_SETUP_PROP_NAME)) {
+    let funcInterceptor = interceptEventListener(listener);
+    if (!funcInterceptor) {
+      return listener; // no need to wrap, just return the original listener
+    }
+
+    type FlowletData = {
+      getTriggerFlowlet: typeof getTriggerFlowlet;
+      callFlowlet: T;
+    };
+
+    const topFlowlet = this.top();
+
+    if (funcInterceptor.testAndSet(IS_FLOWLET_SETUP_PROP_NAME)) {
+      /**
+       * listener is already wrapped and probably has closed on another callFlowlet, or getTriggerFlowlet function
+       * so, we test here to see if we need to create a new interceptor.
+       * This is almost similar to the dependecy list of useCallback in react.
+       */
+      // 
+      const fData = funcInterceptor.getData<FlowletData>(FLOWLET_DATA_PROP_NAME);
+      assert(fData != null, `Flowlet data is not set on the interceptor! This should never happen!`);
+      if (Flags.getFlags().verboseFlowletLogs && fData.getTriggerFlowlet !== getTriggerFlowlet) {
```

Comment:
Super minor,  but notice we check `fData.getTriggerFlowlet !== getTriggerFlowlet` here and immediately below